source/extensions/filters/http/common/jwks_fetcher.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "envoy/api/v2/core/http_uri.pb.h"
+#include "envoy/common/pure.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "jwt_verify_lib/jwks.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+
+class JwksFetcher;
+typedef std::unique_ptr<JwksFetcher> JwksFetcherPtr;
+/**
+ * JwksFetcher interface can be used to retrieve remote JWKS
+ * (https://tools.ietf.org/html/rfc7517) data structures returning a concrete,
+ * type-safe representation. An instance of this interface is designed to
```

Comment:
Ah, you answer the above question here. This is a pretty weird pattern; I would call it out by explicitly naming the class/interface `JwksSingleUseFetcher` or something like that.

Reply:
@htuch I've improved the description of the functionality in 8fa1dc8. I've also made this class re-usable but designed to retrieve a single JWKS at a time.